core/arch/arm/kernel/thread.c

```diff
@@ -613,6 +614,31 @@ size_t thread_stack_size(void)
 	return STACK_THREAD_SIZE;
 }
 
+void get_stack_limits(vaddr_t *start, vaddr_t *end)
+{
+	uint32_t exceptions = thread_mask_exceptions(THREAD_EXCP_FOREIGN_INTR);
+	struct thread_core_local *l = thread_get_core_local();
+	uint32_t cpu_id = get_core_pos();
+	struct thread_ctx *thr = NULL;
+
+	if (l->flags & THREAD_CLF_TMP) {
+		/* We're using the temporary stack for this core */
+		*start = (vaddr_t)stack_tmp[cpu_id];
+		*end = *start + STACK_TMP_SIZE;
+	} else if (l->flags & THREAD_CLF_ABORT) {
+		/* We're using the abort stack for this core */
+		*start = (vaddr_t)stack_abt[cpu_id];
+		*end = *start + STACK_ABT_SIZE;
+	} else if (!l->flags) {
+		/* Were using a thread stack */
+		thr = threads + l->curr_thread;
```

Comment:
Let's assert that `l->curr_thread` is within the expected range.

Reply:
Sure